HOOKS.md

```diff
@@ -8,54 +8,73 @@ The hook receives the data and returns the result via files. Paths to files are
   - Every executable file found in the path is considered a hook.
 
-- The found hooks are sorted alphabetically according to the directories’ and hooks’ names. Then they are executed with the `--config` flag to get bindings to events in JSON format.
+- The found hooks are sorted alphabetically according to the directories’ and hooks’ names. Then they are executed with the `--config` flag to get bindings to events in YAML or JSON format.
 
-- If hook's configuration is successful, the workqueue is filled with `onStartup` hooks.
+- If hook's configuration is successful, the working queue named "main" is filled with `onStartup` hooks.
 
-- After executing `onStartup` hooks, Shell-operator subscribes to Kubernetes events according to configured `kubernetes` bindings.
+- Then, the "main" queue is filled with `kubernetes` hooks with `Synchronization` [binding context](#binding-context) type, so that each hook receives all existing objects described in hook's configuration.
 
-- Then, the work queue is filled with `kubernetes` hooks with `Synchronization` [binding context](#binding-context) type, so that each hook receives all existing objects described in hook's configuration.
+- After executing `kubernetes` hook with `Synchronization` binding context, Shell-operator starts a monitor of Kubernetes events according to configured `kubernetes` binding.
+  - Each monitor stores a *snapshot* — a refreshable list of all Kubernetes objects that match a binding definition.
 
 Next the main cycle is started:
 
-- Hooks are adding to the queue on events:
-  - `kubernetes` hooks are added to the queue from events that occur in Kubernetes,
-  - `schedule` hooks are added according to the schedule.
+- Event handler adds hooks to the named queues on events:
+  - `kubernetes` hooks are added to the queue when desired [WatchEvent](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#watchevent-v1-meta) is received from Kubernetes,
+  - `schedule` hooks are added according to the schedule,
+  - `kubernetes` and `schedule` hooks are added to the "main" queue or to the named queue if `queue` field was specified.
 
-- Queue handler executes hooks strictly sequentially. If hook fails with an error (non-zero exit code), Shell-operator restarts it (every 5 seconds) until success. In case of an erroneous execution of some hook, when other events occur, the queue will be filled with new tasks, but their execution will be blocked until the failing hook succeeds.
+- Each named queue has an own queue handler which executes hooks strictly sequentially. If hook fails with an error (non-zero exit code), Shell-operator restarts it (every 5 seconds) until success. In case of an erroneous execution of some hook, when other events occur, the queue will be filled with new tasks, but their execution will be blocked until the failing hook succeeds.
```

Comment:
```suggestion
- Each named queue has its own queue handler which executes hooks strictly sequentially. If hook fails with an error (non-zero exit code), Shell-operator restarts it (every 5 seconds) until succeeds. In case of an erroneous execution of some hook, when other events occur, the queue will be filled with new tasks, but their execution will be blocked until the failing hook succeeds.
```